### PR TITLE
Fix aggressive abbreviation

### LINKED ISSUE #1185: First banner is abbreviated too aggressively
The first banner that appears when entering turn-by-turn navigation is abbreviated as aggressively as possible, probably because the `availableBounds` calculation occurs before the banner reaches its full width:

s<br>
_This should say “South” but says “S” instead._

/cc @frederoni @JThramer

### PR DESCRIPTION
**Pull request by frederoni**
Fixes #1185 

Fixes aggressive abbreviations by using an RTL and layout state-agnostic width calculation.

@1ec5 👀

### CODE REVIEW

**Comment by 1ec5** [comment]
Also, don’t forget the next banner.

File: MapboxNavigation/InstructionsBannerViewLayout.swift [hunk 1]
```diff
@@ -118,25 +118,15 @@ extension BaseInstructionsBannerView {
...
-            let availableWidth: CGFloat
-            switch UIApplication.shared.userInterfaceLayoutDirection {
-            case .leftToRight:
-                availableWidth = self.bounds.width - self.maneuverView.frame.maxX - 8 * 2
-            case .rightToLeft:
-                availableWidth = self.maneuverView.frame.minX - 8 * 2
-            }
+        primaryLabel.availableBounds = { [unowned self] in
+            // Available width H:|-padding-maneuverView-padding-availableWidth-padding-|
+            let availableWidth = self.bounds.width-BaseInstructionsBannerView.maneuverViewSize.width-BaseInstructionsBannerView.padding*3
```
Comment (JThramer) [inline MapboxNavigation/InstructionsBannerViewLayout.swift]: Can we put some whitespace in this mathematical expression?

File: MapboxNavigation/InstructionsBannerViewLayout.swift [hunk 2]
```diff
@@ -118,25 +118,15 @@ extension BaseInstructionsBannerView {
...
-            let availableWidth: CGFloat
-            switch UIApplication.shared.userInterfaceLayoutDirection {
-            case .leftToRight:
-                availableWidth = self.bounds.width - self.maneuverView.frame.maxX - 8 * 2
-            case .rightToLeft:
-                availableWidth = self.maneuverView.frame.minX - 8 * 2
-            }
+        secondaryLabel.availableBounds = { [unowned self] in
+            // Available width H:|-padding-maneuverView-padding-availableWidth-padding-|
+            let availableWidth = self.bounds.width-BaseInstructionsBannerView.maneuverViewSize.width-BaseInstructionsBannerView.padding*3
```
Comment (JThramer) [inline MapboxNavigation/InstructionsBannerViewLayout.swift]: Here too.

File: MapboxNavigation/NextBannerView.swift [hunk 3]
```diff
@@ -43,14 +43,8 @@ open class NextBannerView: UIView {
...
-        instructionLabel.availableBounds = {
-            let availableWidth: CGFloat
-            switch UIApplication.shared.userInterfaceLayoutDirection {
-            case .leftToRight:
-                availableWidth = self.bounds.width - self.maneuverView.frame.maxX - 16 * 2
-            case .rightToLeft:
-                availableWidth = self.maneuverView.frame.minX - 16 * 2
-            }
+        instructionLabel.availableBounds = { [unowned self] in
+            let availableWidth = self.bounds.width-BaseInstructionsBannerView.maneuverViewSize.width-BaseInstructionsBannerView.padding*3
```
Comment (JThramer) [inline MapboxNavigation/NextBannerView.swift]: One more.

**Comment by frederoni** [comment]
> But does it address #1185?

@1ec5 by using the bounds instead of the frame, it works around the initial zero frame introduced in #1063 and simplifies RTL.
screenshot 2018-03-05 14 15 38

The initial zero frame is being address in #1184

**Pull request merged**